### PR TITLE
Dashboard: Unified Menu Personalization

### PR DESCRIPTION
**Pull request by kean**
Part of #21492

- Implement Shortcuts (or Quick Actions) personalization. The settings are per-site.
- Add "Comments" quick action
- Reader shortcuts: "Stats" are not at the top

https://github.com/wordpress-mobile/WordPress-iOS/assets/1567433/812b72c4-7523-4f27-8d10-efd169a43e77

## To test:

- Open Personalization menu
- Change the settings and make sure the total number of enabled Quick Actions changes
- Return to Dashboard
- Verify that the Quick Actions are updated
- Verify that the height of the Quick Actions section got updated

### Card not Eligible

- Login with a blog that has no access to "Stats" (or update `func isEligible(for blog: Blog) -> Bool` to return `false`)
- Open Dashboard and verify that the "Stats" shortcut is not visible
- Open Personalization screen and verify that the "Stats" shortcut is not visible

## Regression Notes
1. Potential unintended areas of impact: Dashboard & Personalization
2. What I did to test those areas of impact (or what existing automated tests I relied on): manual
3. What automated tests I added (or what prevented me from doing so): n/a

PR submission checklist:

- [x] I have completed the Regression Notes.
- [x] I have considered adding unit tests for my changes.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

UI Changes testing checklist:
- [ ] Portrait and landscape orientations.
- [x] Light and dark modes.
- [x] Fonts: Larger, smaller and bold text.
- [ ] High contrast.
- [ ] VoiceOver.
- [ ] Languages with large words or with letters/accents not frequently used in English.
- [ ] Right-to-left languages. (Even if translation isn’t complete, formatting should still respect the right-to-left layout)
- [ ] iPhone and iPad. 
- [ ] Multi-tasking: Split view and Slide over. (iPad)

### CODE REVIEW

**Comment by kean** [comment]
Converting to draft because the settings need to be site-specific.

**Comment by kean** [comment]
@momo-ozawa this is now ready for review

**Review by momo-ozawa — COMMENTED** [review]
@kean when I toggle a shortcut then switch to a different blog, the dashboard doesn't update the visibility of the shortcut for the new blog.


https://github.com/wordpress-mobile/WordPress-iOS/assets/6711616/ba7aa124-a0fb-4a27-abf8-7cb2278695f1

**Comment by kean** [comment]
@momo-ozawa the settings are site-specific because some of the features (Pages and Stats) are site-specific. The personalization for cards is also site-specific. It's probably not ideal for A8C users but should be OK for everyone else. cc @osullivanchris

**Comment by kean** [comment]
@momo-ozawa I found an issue where personalization doesn't work correctly after changing sites; working on it.

**Comment by kean** [comment]
@momo-ozawa nice catch, thanks, fixed!

File: WordPress/Classes/ViewRelated/Blog/BlogPersonalization/BlogDashboardPersonalizationViewModel.swift [hunk 1]
```diff
@@ -1,9 +1,15 @@
 import SwiftUI
 
 final class BlogDashboardPersonalizationViewModel: ObservableObject {
+    let quickActions: [DashboardPersonalizationQuickActionViewModel]
     let cards: [BlogDashboardPersonalizationCardCellViewModel]
 
-    init(service: BlogDashboardPersonalizationService, quickStartType: QuickStartType) {
+    init(blog: Blog, service: BlogDashboardPersonalizationService, quickStartType: QuickStartType) {
```
Comment (momo-ozawa) [inline WordPress/Classes/ViewRelated/Blog/BlogPersonalization/BlogDashboardPersonalizationViewModel.swift:7]: FYI `BlogDashboardPersonalizationViewModelTests` is failing now with the error: `missing argument for parameter 'blog'`

**Review by momo-ozawa — APPROVED** [review]
Works as described! LGTM once you fix `BlogDashboardPersonalizationViewModelTests`

File: WordPress/Classes/ViewRelated/Blog/BlogPersonalization/BlogDashboardPersonalizationViewModel.swift (hunk 1 above)
Comment (kean) [reply WordPress/Classes/ViewRelated/Blog/BlogPersonalization/BlogDashboardPersonalizationViewModel.swift:7]: Oops, forgot I wrote these – updated.

**Pull request merged**